README.md

```diff
@@ -407,6 +407,165 @@ after a sequence of diffs are handled.
+* `req` _Object (optional)_: a connection context object that can contain information such as cookies or session data that will be available in the [middleware](#middlewares)
+
+Returns a [`Connection`](#class-sharedbconnection).
+
+#### `listen`
+
+```javascript
+backend.listen(stream: Stream, req: Object): Agent;
+```
+
+Registers a `Stream` with the backend.
+
+* `stream` _Stream_: a [`Stream`](https://nodejs.org/api/stream.html) (or `Stream`-like object) that will be used to communicate between the `Agent` and the `Backend`
+* `req` _Object (optional)_: a connection context object that can contain information such as cookies or session data that will be available in the [middleware](#middlewares)
+
+Returns an `Agent`, which is also available in the [middleware](#middlewares).
+
+#### `close`
+
+```javascript
+backend.close(callback: Function): void;
+```
+
+Disconnects ShareDB and all of its underlying services (database, pubsub, etc.).
+
+* `callback` _Function_: a callback with the signature `function (error: Error): void` that will be called once the services have stopped, or with an `error` if at least one of them could not be stopped
+
+#### `use`
+
+```javascript
+backend.use(action: string | string[], middleware: Function): Backend;
+```
+
+Adds [middleware](#middlewares) to the `Backend`.
+
+* `action` _string | string[]_: an action, or array of action names defining when to apply the middleware
+* `middleware` _Function_: a middleware function with the signature `function (context: Object, callback: Function): void;`. See [middleware](#middlewares) for more details
+
+Returns the `Backend` instance.
+
+#### `addProjection`
+
+```javascript
+backend.addProjection(name: string, collection: string, fields: Object): void;
+```
+
+Adds a [projection](#projections).
+
+* `name` _string_: the name of the projection
+* `collection` _string_: the name of the collection on which to apply the projection
+* `fields` _Object_: a declaration of which fields to include in the projection, such as `{ field1: true }`. Defining sub-field projections is not supported.
+
+#### `submit`
+
+```javascript
+backend.submit(agent: Agent, index: string, id: string, op: Object, options: Object, callback: Function): void;
+```
+
+Submits an operation to the `Backend`.
+
+* `agent` _Agent_: connection agent to pass to the middleware
+* `index` _string_: the name of the target collection or projection
+* `id` _string_: the document ID
+* `op` _Object_: the operation to submit
+<!-- TODO: What does this options object do? -->
+* `options` _Object_: ???
+<!-- TODO: Check that I understand this properly -->
+* `callback` _Function_: a callback with the signature `function (error: Error, ops: Object[]): void;`, where `ops` are the ops that were actually submitted to ShareDB after transforming them up to work on the latest version of the document (which may be different to the op that was originally submitted)
```

Comment:
Is this description of the callback correct?

Reply:
Maybe we should just remove this entire `submit` entry? Can we do anything here that we can't do from a client?

Reply:
The ops are actually the ops that were committed between my current document version, and the version in which my own op was committed. These are used to apply to the client to transform the document up, and keep it consistent with the database.

Let's keep the `submit` entry.